assemblyline_core/submission_client.py

```diff
@@ -133,10 +134,17 @@ def submit(self, submission_obj: SubmissionObject, local_files: List = None, com
         if len(submission_obj.files) == 0 and len(local_files) == 0:
             raise SubmissionException("No files found to submit...")
 
-        if submission_obj.params.ttl:
-            expiry = epoch_to_iso(submission_obj.time.timestamp() + submission_obj.params.ttl * 24 * 60 * 60)
-        else:
-            expiry = None
+        # Figure out the expiry for the submission if none was provided
+        if expiry is None:
+            if submission_obj.params.ttl:
+                expiry = epoch_to_iso(submission_obj.time.timestamp() + submission_obj.params.ttl * 24 * 60 * 60)
+
+        # Enforce the max_dtl
+        if self.config.submission.max_dtl > 0:
+            max_expiry = now_as_iso(self.config.submission.max_dtl)
```

Comment:
Shouldn't we multiply max_dtl by 24*60*60 ?

Reply:
yes we should